### ISSUE TITLE
How to retrieve a class' byte[]?

### ISSUE THREAD

**Issue by jdeppe-pivotal** [classgraph/classgraph#253]
Hi. I'm wanting to try to use `ClassGraph` to extract the actual class (and resource) bytes out of my classpath. To that end, I have code something like this:

```
 public byte[] fetchClass(String name) {
    String classResource = name.replace(".", "/") + ".class";
    ResourceList resources = scanResult.getResourcesWithPath(classResource);
    return resources.get(0).load();
}
```

However, it appears that I always end up with a zero-byte array. For a particular class I'm trying to have it extract from a jar file. Debugging a bit it seems that when it gets to actually trying to `load()` the bytes, the `ZipEntry` is invalid (stale?) as nothing gets loaded. Am I doing this wrong?

**Comment by lukehutch** [comment]
Hi, it looks like you're doing the right sort of thing, so this may be a bug. However this exact pattern seems to work for `ClassGraphClassLoader`, if you look at the ClassGraph code. Can you please set up a minimum basic testcase demonstrating the problem?

**Comment by lukehutch** [comment]
Also, please attach verbose logs. Thanks!

**Comment by lukehutch** [comment]
PS don't forget to close the `Resource` after you use it -- see the code I linked, or use one of the `ResourceList` methods starting with `forAll...`, which do this for you.

**Comment by jdeppe-pivotal** [comment]
Turns out I was hitting this on `4.0.6` but when I switched to `4.1.7` all was good.

**Comment by lukehutch** [comment]
Ah, makes sense, I fixed a bug in 4.1.7 that you must have been hitting. (I just released 4.2.3, so maybe bump up to that version now.) Thanks.
